light-curve/light_curve/light_curve_py/features/rainbow/bolometric.py

```diff
@@ -186,7 +195,152 @@ def peak_time(t0, amplitude, rise_time, fall_time):
         return t0 + np.log(fall_time / rise_time) * rise_time * fall_time / (rise_time + fall_time)
 
 
+@dataclass()
+class LinexpBolometricTerm(BaseBolometricTerm):
+    """Linexp function, symmetric form. Generated using a prototype version of Multi-view
+    Symbolic Regression (Russeil et al. 2024, https://arxiv.org/abs/2402.04298) on
+    a SLSN ZTF light curve (https://ztf.snad.space/dr17/view/821207100004043)"""
+
+    @staticmethod
+    def parameter_names():
+        return ["reference_time", "amplitude", "rise_time"]
+
+    @staticmethod
+    def parameter_scalings():
+        return ["time", "flux", "timescale"]
+
+    @staticmethod
+    def value(t, t0, amplitude, rise_time):
+        dt = t0 - t
+        protected_rise = math.copysign(max(1e-5, abs(rise_time)), rise_time)
+
+        # Coefficient to make peak amplitude equal to unity
+        scale = 1 / (protected_rise * np.exp(-1))
+
+        power = -dt / protected_rise
+        power = np.where(power > 100, 100, power)
+        result = amplitude * scale * dt * np.exp(power)
+
+        return result
+
+    @staticmethod
+    def initial_guesses(t, m, sigma, band):
+
+        A = np.ptp(m)
+
+        # Compute points after or before maximum
+        peak_time = t[np.argmax(m)]
+        after = t[-1] - peak_time
+        before = peak_time - t[0]
+
+        # Peak position as weighted centroid of everything above zero
+        idx = m > 0
```

Comment:
It will crash if there is no positive points in the dataset. Better use e.g. `m > np.median(m)` or something else like that

Reply:
Fixed